core/src/main/java/tc/oc/pgm/shops/ShopModule.java

```diff
@@ -0,0 +1,217 @@
+
+  @Override
+  public MatchModule createMatchModule(Match match) {
+    return new ShopMatchModule(match, shops, shopKeepers);
+  }
+
+  @Override
+  public Collection<MapTag> getTags() {
+    return TAGS;
+  }
+
+  public static class Factory implements MapModuleFactory<ShopModule> {
+
+    @Override
+    public ShopModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      ShopModule module = new ShopModule();
+      KitParser kitParser = factory.getKits();
+      SetMultimap<String, KeeperData> locations = parseShopKeepers(doc.getRootElement(), logger);
+
+      for (Element shop : XMLUtils.flattenElements(doc.getRootElement(), "shops")) {
+        Attribute shopName = XMLUtils.getRequiredAttribute(shop, "name");
+        List<Category> categories = Lists.newArrayList();
+
+        for (Element category : XMLUtils.getChildren(shop, "category")) {
+          Attribute categoryName = XMLUtils.getRequiredAttribute(category, "name");
+          Attribute categoryMaterial = XMLUtils.getRequiredAttribute(category, "material");
+          Material material = parseMaterial(categoryMaterial, category);
+          List<Icon> icons = parseIcons(category, kitParser, logger);
+
+          categories.add(new Category(categoryName.getValue(), material, icons));
+        }
+
+        if (categories.isEmpty()) {
+          throw new InvalidXMLException("At least one <category> is required per shop", shop);
+        }
+
+        Shop shopInstance = new Shop(shopName.getValue(), categories);
+        factory.getFeatures().addFeature(shop, shopInstance);
+        module.shops.add(shopInstance);
+
+        Set<KeeperData> shopKeeperLocations = locations.get(shopName.getValue());
+        if (shopKeeperLocations.isEmpty()) {
+          throw new InvalidXMLException("At least one <shopkeeper> is required per shop", shop);
+        }
+        shopKeeperLocations.stream()
+            .map(
+                data ->
+                    new ShopKeeper(
+                        data.getName(),
+                        data.getLocation(),
+                        data.getYaw(),
+                        data.getPitch(),
+                        data.getType(),
+                        shopInstance))
+            .forEach(keeper -> module.shopKeepers.add(keeper));
+      }
+
+      return !module.shops.isEmpty() ? module : null;
+    }
+  }
+
+  private static SetMultimap<String, KeeperData> parseShopKeepers(Element root, Logger logger)
+      throws InvalidXMLException {
+    SetMultimap<String, KeeperData> shopkeepers = HashMultimap.create();
+
+    for (Element shopkeeper : XMLUtils.flattenElements(root, "shopkeepers")) {
+      Attribute locationAttr = XMLUtils.getRequiredAttribute(shopkeeper, "location");
+      Attribute yawAttr = XMLUtils.getRequiredAttribute(shopkeeper, "yaw");
+      Attribute pitchAttr = XMLUtils.getRequiredAttribute(shopkeeper, "pitch");
```

Comment:
This is essentially what a `PointProvider` gives, look into `PointParser` and just parse this as a child, similar to how spawns do it, it will save you the head-ache of having to parse location, yaw, pitch yourself, and also provide more flexibility: eg angle (which faces the location provided) or spawning the shop in a "random position within the region" capability. Usually, this will just be a <point>x,y,z</point> tho. Another example that uses this is flag posts, which also use a point provider (and can spawn in a whole region, or a union of points, but generally just use a single point)